src/chatClient.h

```diff
@@ -82,7 +82,7 @@ class ChatRoom: public chatd::Listener, public DeleteTrackable
     /** @endcond PRIVATE */
 
     /** @brief The text that will be displayed on the chat list for that chat */
-    virtual const std::string& titleString() const = 0;
+    virtual std::string titleString() const = 0;
```

Comment:
This change means we will be copying the `std::string` at every call without a good reason. Better to keep a reference or change to a `const char*` if the `string` object is invalid out of the scope of the function.